vllm/v1/metrics/loggers.py

```diff
@@ -346,32 +347,63 @@ def __init__(self, vllm_config: VllmConfig, engine_index: int = 0):
             self.labelname_running_lora_adapters = "running_lora_adapters"
             self.max_lora = vllm_config.lora_config.max_loras
             self.gauge_lora_info = \
-                prometheus_client.Gauge(
+                self._create_gauge(
                     name="vllm:lora_requests_info",
                     documentation="Running stats on lora requests.",
                     labelnames=[
                         self.labelname_max_lora,
                         self.labelname_waiting_lora_adapters,
                         self.labelname_running_lora_adapters,
-                    ])
+                    ],
+                    multiprocess_mode="livemostrecent"
+                )
+
+    def _create_gauge(self,
+                      name: str,
+                      documentation: Optional[str],
+                      labelnames: list[str],
+                      multiprocess_mode: str = "all"):
+        return prometheus_client.Gauge(name=name,
+                                       documentation=documentation,
+                                       labelnames=labelnames,
+                                       multiprocess_mode=multiprocess_mode)
+
+    def _create_counter(self, name: str, documentation: Optional[str],
+                        labelnames: list[str]):
+        return prometheus_client.Counter(name=name,
+                                         documentation=documentation,
+                                         labelnames=labelnames)
```

Comment:
What's the purpose of adding this indirection if all of the args are just passed to the corresponding constructors?

Reply:
Mostly modularity. We can extend these, for example in this PR https://github.com/vllm-project/vllm/pull/17925 we want to wrap these primitives with their Ray equivalent.

Reply:
It might make more sense to make this particular change in that other PR then since it's not directly related to this one.